sphinx_gallery/gen_gallery.py

```diff
@@ -843,29 +863,37 @@ def get_entry_type(entry):
             assert entry in gallery_conf['api_entries']['function']
             return 'func'
 
+    workdirs = [os.path.join(app.builder.srcdir, examples_dir)
+                for examples_dir in gallery_conf['examples_dirs']]
+
     # find used and unused API entries
     unused_api_entries = list()
     used_api_entries = dict()
     for entry in example_files:
         # don't include built-in methods etc.
         if re.match(gallery_conf['api_usage_ignore'], entry) is not None:
             continue
-        # check if backreferences empty
-        example_fname = os.path.join(
-            backreferences_dir, f'{entry}.examples.new')
-        if not os.path.isfile(example_fname):  # use without new
-            example_fname = os.path.splitext(example_fname)[0]
-        assert os.path.isfile(example_fname)
-        if os.path.getsize(example_fname) == 0:
-            unused_api_entries.append(entry)
-        else:
+        # just search for function/class/method, this may lead to a few
+        # false-positives for functions/classes/methods with the same
+        # name, but for this purpose, that is preferable to false-negatives
```

Comment:
>  if you've seen one fit, you've kind of seen them all

I'm not sure, even in `sklearn` the `kwargs` there can vary a bit. A more drastic example of where this is problematic is `inst.plot` in MNE-Python -- we have many of them (dozens?) and if make a new class with a `.plot` method and *don't* have an example for it, we'd want to know

Reply:
Yeah, I suppose it's better to wait for the right way to do this by following the variable types in code execution that is much harder to do. I will change this to just the fixes and then it can be merged.